sys/riscv/riscv/trap.c

```diff
@@ -197,22 +197,11 @@ cpu_fetch_syscall_args(struct thread *td)
 
 #include "../../kern/subr_syscall.c"
 
-/*
- * This cannot use _CHERI_PRINTF_CAP_ARG due to the casts (the trapframe stores
- * uintcap_t rather than void * __capability).
- */
 #if __has_feature(capabilities)
-#ifdef __CHERI_PURE_CAPABILITY__
-#define	PRINT_REG_ARG(value)	((void * __capability)(value))
-#else
-#define	PRINT_REG_ARG(value)	((void * __capability *)&(value))
-#endif
-#define PRINT_REG(name, value)					\
-	printf(name " = " _CHERI_PRINTF_CAP_FMT "\n",		\
-	    PRINT_REG_ARG(value));
-#define PRINT_REG_N(name, n, array)				\
-	printf(name "[%d] = " _CHERI_PRINTF_CAP_FMT "\n", n,	\
-	    PRINT_REG_ARG((array)[n]));
+#define PRINT_REG(name, value)	\
+	printf(name " = %#.16lp\n", (void * __capability)value);
+#define PRINT_REG_N(name, n, array)	\
+	printf(name "[%d] = %#.16lp\n", n, (void * __capability)(array)[n]);
 #else
 #define PRINT_REG(name, value)	printf(name " = 0x%016lx\n", value)
```

Comment:
If we had a macro for a padded `(void * __capability)` format (`%#.16lp` in hybrid/purecap, `%#016lx` in non-CHERI), we could avoid this `#ifdef`. Should also be useful for arm64/trap.c and mips/trap.c.

Reply:
Well one's taking a `uint64_t` and one's taking a `void * __capability` so both the format string and argument would need something given in the capability case we need a cast from `uintcap_t` to `void * __capability`.